### PR TITLE
Source Google Search Console: refactor spec

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolving https://github.com/airbytehq/airbyte/issues/23571

## How
Refactor Specs
Right now we expect `custom_reports`  as `str` in format 
```
{"name": "<report-name>", "dimensions": ["<dimension-name>", ...]} 
```
image

For better UX we change `custom_reports` to `list` to add reports easier:

image

image

Apply to migrate in DB:
```
CREATE FUNCTION migrate_google_search_console_config()
RETURNS VOID AS
$$
BEGIN
	ALTER TABLE actor ADD COLUMN custom_reports_text text;
update actor a set custom_reports_text = coalesce(nullif(configuration->>'custom_reports', ''), '[]') where actor_definition_id='eb4c9e00-db83-4d63-a386-39cfa91012a8';
update actor a set configuration = jsonb_set(configuration, '{custom_reports}', coalesce(custom_reports_text::jsonb, '[]'::jsonb), true) where actor_definition_id='eb4c9e00-db83-4d63-a386-39cfa91012a8'; 
 ALTER TABLE actor DROP COLUMN custom_reports_text;
END;
$$
LANGUAGE plpgsql;

SELECT migrate_google_search_console_config();
```

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
custom_reports field type changed from `str` to `List`. Breaking changes in spec.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.py:1]: shouldn't we drop spec.json?

**Review by davydov-d — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+from datetime import date
+from enum import Enum
+from typing import List, Optional, Union
+
+from airbyte_cdk.models import AdvancedAuth, AuthFlowType, OAuthConfigSpecification
+from airbyte_cdk.sources.config import BaseConfig
+from pydantic import BaseModel, Field
+from source_google_search_console.streams import SearchAnalyticsByCustomDimensions
+
+logger = logging.getLogger("airbyte")
```
[the remark this answers is not in the document]
Comment (artem1205) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.py]: updated

File: airbyte-integrations/connectors/source-google-search-console/Dockerfile [hunk 3]
```diff
@@ -12,5 +12,5 @@ RUN pip install .
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=0.1.20
+LABEL io.airbyte.version=0.1.21
```
[the remark this answers is not in the document]
Comment (artem1205) [reply airbyte-integrations/connectors/source-google-search-console/Dockerfile]: updated

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.py:1]: json was moved to `integration_tests` folder

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py [hunk 4]
```diff
@@ -167,3 +170,12 @@ def get_authenticator(self, config):
                 service_account_info=authorization["service_account_info"],
                 email=authorization["email"],
             )
+
+    def spec(self, logger: logging.Logger) -> ConnectorSpecification:
+        return ConnectorSpecification(
+            documentationUrl="https://docs.airbyte.com/integrations/sources/google-search-console",
+            changelogUrl="https://docs.airbyte.com/integrations/sources/google-search-console",
+            supported_destination_sync_modes=[DestinationSyncMode.append],
```
[the remark this answers is not in the document]
Comment (artem1205) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py:178]: WDYM ?

File: airbyte-integrations/connectors/source-google-search-console/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-google-search-console/Dockerfile]: I meant it should be 1.0.0 :)

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py:178]: I mean you're setting just one supported destination sync mode - "append". How about "overwrite"?

[the remark this answers is not in the document]
Comment (artem1205) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py]: deleted, as optional field

**Review by davydov-d — APPROVED** [review]

**Comment by artem1205** [comment]
@sherifnada , please take a look at PR and psql script in description.

**Comment by lmossman** [comment]
FYI I left another comment about this change on the related slack thread: https://airbytehq-team.slack.com/archives/C02UF50V9HA/p1677869635655799?thread_ts=1677685710.769469&cid=C02UF50V9HA

**Comment by lazebnyi** [comment]
Closed as deprioritized.

**Pull request closed without merging**